lib/resources/package.rb

```diff
@@ -133,6 +164,30 @@ def info(package_name)
         type: 'rpm',
       }
     end
+
+    private
+
+    def validate_arguments
```

Comment:
It feels like this method and the one below it are unnecessary layers of abstraction. Any reason not just to fold these into `missing_requirements` in the `Rpm` class? If we collect additional requirements later, we can break those down until separate methods but it feels like a bunch of hopping around between methods to get to a simple "if this parameter isn't nil, check to make sure it's actually a file on disk" check.